contracts/src/TeleporterTokenDestination.sol

```diff
@@ -199,93 +199,13 @@ abstract contract TeleporterTokenDestination is
      * - {amount} must be greater than 0
      */
     function _send(SendTokensInput calldata input, uint256 amount) internal sendNonReentrant {
-        require(input.recipient != address(0), "TeleporterTokenDestination: zero recipient address");
-        require(input.requiredGasLimit > 0, "TeleporterTokenDestination: zero required gas limit");
-
-        uint256 primaryFee;
-        (amount, primaryFee) = _prepareSend({
-            destinationBlockchainID: input.destinationBlockchainID,
-            destinationBridgeAddress: input.destinationBridgeAddress,
-            amount: amount,
-            primaryFeeTokenAddress: input.primaryFeeTokenAddress,
-            primaryFee: input.primaryFee,
-            secondaryFee: input.secondaryFee
-        });
+        _validateSendTokensInput(input);
 
-        // If the destination blockchain is the source blockchain,
-        // no multi-hop is needed. Only the required gas limit for the Teleporter message back to
-        // {sourceBlockchainID} is needed, which is provided by {input.requiredGasLimit}.
-        // Else, there will be a multi-hop transfer to the final destination.
-        // The first hop back to {sourceBlockchainID} requires {MULTI_HOP_REQUIRED_GAS},
-        // and the second hop to the final destination requires {input.requiredGasLimit}.
-        BridgeMessage memory message;
-        uint256 messageRequiredGasLimit = input.requiredGasLimit;
         if (input.destinationBlockchainID == sourceBlockchainID) {
-            // If the destination blockchain is the source bridge instance's blockchain,
-            // the destination bridge address must match the token source address,
-            // and no secondary fee or fallback is needed.
-            require(
-                input.destinationBridgeAddress == tokenSourceAddress,
-                "TeleporterTokenDestination: invalid destination bridge address"
-            );
-            require(input.secondaryFee == 0, "TeleporterTokenDestination: non-zero secondary fee");
-            require(
-                input.multiHopFallback == address(0),
-                "TeleporterTokenDestination: non-zero multi-hop fallback"
-            );
-            message = BridgeMessage({
-                messageType: BridgeMessageType.SINGLE_HOP_SEND,
-                payload: abi.encode(SingleHopSendMessage({recipient: input.recipient, amount: amount}))
-            });
+            _processSend(input, amount);
```

Comment:
I'd be slightly in favor of moving the calls to `_validateMultiHopInput` and `_validateSingleHopInput` to this function. Reading this function as is it would be easy to conclude that all of the input validation happens in the call to `_validateSendTokensInput` above, and that the `_processSend` functions expect the input to be validated already. I do think it's reasonable for `_processSend` and _processSendMultihop` to expect the input to be valid.

Reply:
I think I'm slightly opposed to the idea. I think functions should usually validate their own input, but maybe that's just because I'm bad at reading the documentation 😓

Reply:
I don't feel strongly either way, but that's a good point that functions should validate their own inputs.